pkg/config/loader/config_loader.go

```diff
@@ -23,6 +23,7 @@ import (
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/files"
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/maps"
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/slices"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/config"
```

Comment:
nit: importing the parent package always feels a bit iffy to me, and like it will end with a cyclic dependency at some point

Reply:
I think, in this case, it's absolutely fine. If the config in any case depends on config-loading or config-writing, we did something absolutely wrong.
We could also move it to a completely different set of packages, away from the config. 

Similar case for the parameters

Reply:
> If **the config** in any case depends on config-loading or config-writing, we did something absolutely wrong.

That depends fully on what the "config" package actually is.
I'd argue that especially with sub-packages it can be understood as "all things config related", rather than "the config type". 

I agree that there is something very wrong if the config type ever depends on the loader.
But I can see how someone taking a casual look at this a few months from now or new to the code-base could end up adding something that causes a cycle/breaks current understanding. 

I've marked it as a nitpick because I agree it's fine in this case and can't think of a much better way. A config/type pkg makes the use of Config strange to read (type.Config)...
A fully separate config_loader package might actually not be too bad now that I think of it though?